### PR TITLE
fix: remove make-dir package and use `fs/promises`

### PR DESCRIPTION
**Pull request by XhmikosR**
**- Summary**

**- Test plan**

- CI is green

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: src/utils/fs.ts [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { lstat, readdir, readFile, stat, unlink, writeFile } from 'fs'
```
Comment (XhmikosR) [inline src/utils/fs.ts:1]: I'm unsure why the direct `fs/promises` imports are not used here in this file. Would be nice to get rid of `promisify` if the same can be done with `fs/promises`

**Comment by XhmikosR** [comment]
This should be ready now assuming CI is green.

File: src/utils/fs.ts [hunk 2]
```diff
@@ -88,29 +78,27 @@ const resolveFunctionsDirectories = (input: string | string[]) => {
...
   resolveFunctionsDirectories,
   safeUnlink,
-  pStat as stat,
-  pWriteFile as writeFile,
-  pReadFile as readFile,
   mkdirAndWriteFile,
 }
 export type { FsCache }
+
+export { stat, lstat, writeFile, readFile } from 'fs/promises'
```
Comment (XhmikosR) [inline src/utils/fs.ts:104]: Not sure about this one. It doesn't make any sense to have these now.

**Comment by XhmikosR** [comment]
All right, I need to fix the error on Node.js 12.20.0. It doesn't support `fs/promises`. Any suggestions how to tackle the issue?

**Comment by XhmikosR** [comment]
How about this solution?

```js
import { promises as fs } from 'fs'

// foo

fs.writeFile()
```

File: src/utils/fs.ts [hunk 3]
```diff
@@ -88,29 +78,25 @@ const resolveFunctionsDirectories = (input: string | string[]) => {
   return absoluteDirectories
 }
 
-const mkdirAndWriteFile: typeof pWriteFile = async (path, ...params) => {
+const mkdirAndWriteFile: typeof fs.writeFile = async (path, ...params) => {
```
Comment (XhmikosR) [inline src/utils/fs.ts:81]: Not familiar with TS, I could use a confirmation

File: src/manifest.ts [hunk 4]
```diff
@@ -1,8 +1,8 @@
+import { promises as fs } from 'fs'
```
Comment (XhmikosR) [inline src/manifest.ts:1]: Maybe it'll be better and more consistent with your other projects to use `import { promises as fsPromises } from 'fs'`?

**Comment by XhmikosR** [comment]
It seems here the supported versions differ from the ones in package.json: https://github.com/netlify/zip-it-and-ship-it/blob/10ac495f798acc62fe46c24dbcfaa197b5be889e/src/config.ts#L8

Not sure if they should be updated?

**Review by netlify-team-account-1 — APPROVED** [review]

**Pull request merged**